### DISCUSSION TITLE
Syntax error highlighting unhelpful within R expression

### DISCUSSION THREAD

**Discussion by pedroliman** [posit-dev/positron#4177] [Feedback]
I accidentally forgot a parenthesis within a with statement, and the whole script is marked with syntax error highlighting. Here's a way to reproduce this with a simple script.

<img width="727" alt="image" src="https://github.com/user-attachments/assets/683ce1ab-229b-455f-a3ba-5547f06b2977">

If I close the parenthesis, all issues go away as they should. And if my parenthesis is missing outside of the expression block, it works as expected, highlighting only the part that is wrong:

<img width="727" alt="image" src="https://github.com/user-attachments/assets/157e6b90-14de-4d55-ad88-116a0ceaa039">

Code:
```r

# Let's say we want to use with to change things in a list (not my favorite way of doing things, but...)
asd = list()

asd$abc = 2

# If I 
with(

  asd,

  {
  
  # If I forget this closing parenthesis, the whole with gets marked.
  # that's not very helpful if the with block is a little long.
    a = sum(abc * 2)

  }

  
)

a = sum(abc * 2

b = 123

```

Is this a bug?

**Comment by DavisVaughan** [comment]
Already tracked by https://github.com/posit-dev/positron/issues/2943, but this will be another useful test case thanks!
